cmake/Dependencies.cmake

```
@@ -1835,51 +1835,31 @@ if(USE_KINETO)
   message(STATUS "  KINETO_LIBRARY_TYPE = ${KINETO_LIBRARY_TYPE}")
 
   if(NOT LIBKINETO_NOCUPTI)
-    set(CUDA_SOURCE_DIR "${CUDA_TOOLKIT_ROOT_DIR}" CACHE STRING "")
-    message(STATUS "  CUDA_SOURCE_DIR = ${CUDA_SOURCE_DIR}")
-    message(STATUS "  CUDA_INCLUDE_DIRS = ${CUDA_INCLUDE_DIRS}")
-
-    if(NOT MSVC)
-      if(USE_CUPTI_SO)
-        set(CUPTI_LIB_NAME "libcupti.so")
-      else()
-        set(CUPTI_LIB_NAME "libcupti_static.a")
-      endif()
+    if(USE_CUPTI_SO)
+      set(CUPTI_LIB CUDA::cupti)
     else()
-      set(CUPTI_LIB_NAME "cupti.lib")
-    endif()
-
-    find_library(CUPTI_LIBRARY_PATH ${CUPTI_LIB_NAME} PATHS
-        ${CUDA_SOURCE_DIR}
-        ${CUDA_SOURCE_DIR}/extras/CUPTI/lib64
-        ${CUDA_SOURCE_DIR}/lib
-        ${CUDA_SOURCE_DIR}/lib64
-        NO_DEFAULT_PATH)
-
-    find_path(CUPTI_INCLUDE_DIR cupti.h PATHS
-        ${CUDA_SOURCE_DIR}/extras/CUPTI/include
-        ${CUDA_INCLUDE_DIRS}
-        ${CUDA_SOURCE_DIR}
-        ${CUDA_SOURCE_DIR}/include
-        NO_DEFAULT_PATH)
-
-    if(CUPTI_LIBRARY_PATH AND CUPTI_INCLUDE_DIR)
-      message(STATUS "  CUPTI_INCLUDE_DIR = ${CUPTI_INCLUDE_DIR}")
-      set(CUDA_cupti_LIBRARY ${CUPTI_LIBRARY_PATH})
+      set(CUPTI_LIB CUDA::cupti_static)
+    endif()
+
+    if(TARGET ${CUPTI_LIB})
+      message(STATUS "  CUPTI_INCLUDE_DIR = ${CUDAToolkit_CUPTI_INCLUDE_DIR}")
+      get_target_property(CUDA_cupti_LIBRARY ${CUPTI_LIB} IMPORTED_LOCATION)
       message(STATUS "  CUDA_cupti_LIBRARY = ${CUDA_cupti_LIBRARY}")
       message(STATUS "Found CUPTI")
       set(LIBKINETO_NOCUPTI OFF CACHE STRING "" FORCE)
 
       # I've only tested this sanity check on Linux; if someone
       # runs into this bug on another platform feel free to
       # generalize it accordingly
-      if(NOT USE_CUPTI_SO AND UNIX)
+      if(NOT USE_CUPTI_SO)
         include(CheckCXXSourceRuns)
         # rt is handled by the CMAKE_REQUIRED_LIBRARIES set above
-        if(NOT APPLE)
+        if(UNIX)
```

This needs to exclude Apple (though CUDA is dead on MacOS), as `whole-archive` is not a correct option
```suggestion
        if(UNIX and NOT APPLE)
```